projects/storefrontlib/cms-components/content/paragraph/paragraph.component.ts

```diff
@@ -25,16 +25,33 @@ export class ParagraphComponent {
       const element = event.target as HTMLAnchorElement;
       const href = element?.getAttribute('href');
 
-      // Use router for internal link navigation
-      if (href?.indexOf('/') === 0) {
-        event.preventDefault();
-        this.router.navigate([`/${href}`]);
+      /**
+       * TODO: (#CXSPA-778) Remove feature config check and deprecated navigation method in 6.0.
+       */
+      if (this.featureConfigService?.isLevel('5.1')) {
+        const documentHost =
+          element.ownerDocument.URL.split('://')[1].split('/')[0];
+
+        // Use router for internal link navigation
+        if (href && documentHost === element.host) {
+          event.preventDefault();
+          this.router.navigateByUrl(href);
+        }
+      } else {
+        if (href?.indexOf('/') === 0) {
+          event.preventDefault();
+          this.router.navigate([`/${href}`]);
+        }
       }
     }
   }
 
+  /**
+   * TODO: (#CXSPA-778) Remove featureConfigService from constructor in 6.0.
+   */
   constructor(
     public component: CmsComponentData<CmsParagraphComponent>,
-    protected router: Router
+    protected router: Router,
+    protected featureConfigService?: FeatureConfigService
```

Comment:
`should:` Please double check if it works without the `Optional()` decorator, as we want to handle optional dependency injection